lib/insight.js

```diff
@@ -64,7 +64,11 @@ Insight.prototype.getUnspentUtxos = function(addresses, callback) {
     if (err || res.statusCode !== 200) {
       return callback(err || res);
     }
-    unspent = _.map(unspent, UnspentOutput);
+    try {
+      unspent = _.map(unspent, UnspentOutput);
+    } catch (ex) {
+      return callback(ex);
+    }
```

Comment:
can you check for a specific error type here and throw the error if it isn't?

e.g:
```
catch (ex) {
  if (ex instanceof bitcore.errors) {
    return callback(ex);
  }
  throw ex;
}
```